src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -25,32 +25,56 @@
 namespace Microsoft.ML.Data
 {
     /// <summary>
-    /// Loads a text file into an IDataView. Supports basic mapping from input columns to IDataView columns.
+    /// Loads a text file into an IDataView. Supports basic mapping from input columns to <see cref="IDataView"/> columns.
     /// </summary>
     public sealed partial class TextLoader : IDataReader<IMultiStreamSource>, ICanSaveModel
     {
-        /// <example>
-        /// Scalar column of <seealso cref="DataKind"/> I4 sourced from 2nd column
-        ///      col=ColumnName:I4:1
-        ///
-        /// Vector column of <seealso cref="DataKind"/> I4 that contains values from columns 1, 3 to 10
-        ///     col=ColumnName:I4:1,3-10
-        ///
-        /// Key range column of KeyType with underlying storage type U4 that contains values from columns 1, 3 to 10, that can go from 1 to 100 (0 reserved for out of range)
-        ///     col=ColumnName:U4[100]:1,3-10
-        /// </example>
+        /// <summary>
+        /// Describes how an input column should be mapped to an <see cref="IDataView"/> column.
+        /// </summary>
         public sealed class Column
         {
+            // Examples of how a column is defined in command line API:
+            // Scalar column of <seealso cref="DataKind"/> I4 sourced from 2nd column
+            //      col=ColumnName:I4:1
+            // Vector column of <seealso cref="DataKind"/> I4 that contains values from columns 1, 3 to 10
+            //     col=ColumnName:I4:1,3-10
+            // Key range column of KeyType with underlying storage type U4 that contains values from columns 1, 3 to 10, that can go from 1 to 100 (0 reserved for out of range)
+            //     col=ColumnName:U4[100]:1,3-10
+
+            /// <summary>
+            /// Describes how an input column should be mapped to an <see cref="IDataView"/> column.
+            /// </summary>
             public Column() { }
 
+            /// <summary>
+            /// Describes how an input column should be mapped to an <see cref="IDataView"/> column.
+            /// </summary>
+            /// <param name="name">Name of the column.</param>
+            /// <param name="type"><see cref="DataKind"/> of the items in the column. If <see langword="null"/> defaults to a <see cref="DataKind.R4"/>.</param>
```

Comment:
>If <see langword="null"/> [](start = 86, length = 25)

Just a rant.
Only reason why we have it is CommandLine in which you can say something like `col=ColA:0" and expect it to work.
On programmatic level idea of "Hey its a nullable field but if you set it null we will treat it as float" is weird.
If you want to change it, one tricky place is TryParse method line 123, just set Type straight to DataKind.Num since in other cases it can be never setup.

Reply:
I agree it does not make much sense to have this as a nullable type. We should just have a default. I would like to address this in another PR as I mention in my other comment.

---
In reply to: [255805797](https://github.com/dotnet/machinelearning/pull/2417#discussion_r255805797) [](ancestors = 255805797)